src/python/pants/backend/python/util_rules/pex_cli.py

```diff
@@ -59,42 +61,48 @@ def default_known_versions(cls):
 @frozen_after_init
 @dataclass(unsafe_hash=True)
 class PexCliProcess:
-    argv: Tuple[str, ...]
+    subcommand: tuple[str, ...]
+    extra_args: tuple[str, ...]
     description: str = dataclasses.field(compare=False)
     additional_input_digest: Optional[Digest]
     extra_env: Optional[FrozenDict[str, str]]
     output_files: Optional[Tuple[str, ...]]
     output_directories: Optional[Tuple[str, ...]]
+    set_resolve_args: bool
     python: Optional[PythonExecutable]
     level: LogLevel
     cache_scope: ProcessCacheScope
 
     def __init__(
         self,
         *,
-        argv: Iterable[str],
+        subcommand: Iterable[str],
+        extra_args: Iterable[str],
         description: str,
         additional_input_digest: Optional[Digest] = None,
         extra_env: Optional[Mapping[str, str]] = None,
         output_files: Optional[Iterable[str]] = None,
         output_directories: Optional[Iterable[str]] = None,
+        set_resolve_args: bool = True,
         python: Optional[PythonExecutable] = None,
         level: LogLevel = LogLevel.INFO,
         cache_scope: ProcessCacheScope = ProcessCacheScope.SUCCESSFUL,
     ) -> None:
-        self.argv = tuple(argv)
+        self.subcommand = tuple(subcommand)
+        self.extra_args = tuple(extra_args)
         self.description = description
         self.additional_input_digest = additional_input_digest
         self.extra_env = FrozenDict(extra_env) if extra_env else None
         self.output_files = tuple(output_files) if output_files else None
         self.output_directories = tuple(output_directories) if output_directories else None
+        self.set_resolve_args = set_resolve_args
```

Comment:
Since the value of this flag needs to be aligned with the `subcommand` that is used, it might be good to put them closer together in the constructor params.